tools/BuildAssets/BuildTasks/Microsoft.Azure.Sdk.Build.Tasks/SdkBuildTasks/ArchiveSymbolsTask.cs

```diff
@@ -0,0 +1,522 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+namespace Microsoft.Azure.Sdk.Build.Tasks
+{
+    using Microsoft.Azure.Sdk.Build.Tasks.BaseTasks;
+    using Microsoft.Build.Framework;
+    using System;
+    using System.Collections.Generic;
+    using System.Diagnostics;
+    using System.IO;
+    using System.Linq;
+    using System.Text;
+
+    /// <summary>
+    /// Archive Symbols Task will
+    /// 1) Copy the build output to an UNC Path (provided from Jenkins Job in order to hardcode paths and to avoid publishing internal UNC paths)
+    /// 2) Create the required request file
+    /// 3) Execute the request to queue up the Symbols Archiving request
+    /// 
+    /// In case the Archive requests and you ever need to queuen up a new request for symbol archive
+    /// Use the cmdline printed in the log (requestOutput\requestOutput.txt) to execute from command line to queue up a new request for the same exact symbols
+    /// 
+    /// TODO:
+    ///     Symbols service has no automated way to inform if the request was processed
+    ///     Currently an email is sent
+    /// </summary>
+    public class ArchiveSymbolsTask : NetSdkTask
+    {
+        #region const
+        const string OUTPUT_ROOTDIRNAME_DEBUG = "Debug";
+        const string OUTPUT_ROOTDIRNAME_RELEASE = "Release";
+        #endregion
+
+        #region fields
+        private string _symbolsArchiveRootDir;
+        private string _buildJobId;
+        private List<string> _assemblyList;
+        #endregion
+
+        protected override INetSdkTask TaskInstance => this;
+
+        /// <summary>
+        /// List of ITaskItem that represents built assembly files        
+        /// </summary>
+        [Required]
+        public ITaskItem[] BuiltAssemblyFileCollection { get; set; }
+
+        /// <summary>
+        /// UNC path where symbols/assemblies will be copied which is used by the Symbols Archive process that will process your archive requests at a later stage
+        /// </summary>
+        [Required]
+        public string ArchiveSymbolsRootDir
+        {
+            get
+            {
+                if (!Directory.Exists(_symbolsArchiveRootDir))
+                {
+                    string alternatePath = System.Environment.GetFolderPath(Environment.SpecialFolder.MyDocuments);
```

Comment:
Why my documents? Wouldn't it be better to use the temp directory?

Reply:
@weshaggard maybe I need more guidance here.
As I will be moving to .NET Core build assemblies, I wanted to use as much as built in APIs/enums.
I could not find a way to get to the temp directory for non-windows platform. The only way I know is using System.Environment.GetEnvironmentVariable("temp"), and I am not sure if this will work on non-windows. And hence thought MyDocuments would be a better choice.
Ideally this will come into play mostly while running tests.

Reply:
For that use `Path.GetTempPath`